include/posix/sys/types.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __POSIX_TYPES_H__
+#define __POSIX_TYPES_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_NEWLIB_LIBC
+#include_next <sys/types.h>
+#endif	/* CONFIG_NEWLIB_LIBC */
+
+#ifdef CONFIG_PTHREAD_IPC
+/* Mutex */
+typedef struct pthread_mutex {
```

Comment:
Can someone explain why this is not simply `typedef k_sem pthread_mutex_t;`?
Current implementation requires a k_sem instance to be created too, while it also pollutes application namespace defining `struct pthread_mutex`.
Same applies for other typedefs.